test/integration/lambda_theory.jl

```diff
@@ -108,35 +110,48 @@ subst_prop = @theory v e a b then alt guard begin
 end
 
 
+isfree(eclass, id) = id ∈ getdata(eclass)
+
+function fresh_var_generator()
+  idx = 0
+  function generate()
+    idx += 1
+    Symbol("a$idx")
+  end
+end
+
+freshvar = fresh_var_generator()
+
 subst_elim = @theory v e c v1 v2 body begin
   # let-const 
   Let(v, e, c::Any) --> c
   # let-Variable-same 
   Let(v1, e, Variable(v1)) --> e
   # TODO fancy let-Variable-diff 
-  Let(v1, e, Variable(v2)) => if addexpr!(_egraph, v1) != addexpr!(_egraph, v2)
-    :(Variable($v2))
-  else
-    _lhs_expr
-  end
+  Let(v1, e, Variable(v2)) => v1 == v2 ? e : Variable(v2)
   # let-lam-same 
   Let(v1, e, λ(v1, body)) --> λ(v1, body)
   # let-lam-diff #TODO captureavoid
-  Let(v1, e, λ(v2, body)) => if v2.id ∈ getdata(e, :freevar, Set()) # is free
-    :(λ($fresh, Let($v1, $e, Let($v2, Variable($fresh), $body))))
+  Let(v1, e, λ(v2, body)) => if isfree(e,v2)
+    fresh = freshvar()
+    λ(fresh, Let(v1, e, Let(v2, Variable(fresh), body)))
```

Comment:
I'm not sure where the `fresh` variable came from before. this one generated from the closure might be a bit ugly?